### ISSUE TITLE
Wrong hostname in some Invite links if using different Server URL

### ISSUE THREAD

**Issue by streetcoder-pm** [jitsi/jitsi-meet#5862]
## Description
I changed the settings of the mobile app to use a different server URL e.g. "meet.neanderfunk.de" opened a new room e.g. "roomname" and invited another person via a text chat app. 

Technically the app is working but the other person has difficulties choosing the correct link because not all links in the default sharing text are replaced correctly according to the configured server URL.

---

## Current behavior
I have the german mobile app therefore the text is in german, but it's about the links placed in the invite text.

```
Klicken Sie auf den folgenden Link, um dem Meeting beizutreten:
https://meet.jit.si/roomname Wollen Sie sich nur auf Ihrem Telefon einwählen?

}Klicken Sie auf diesen Link, um die eingewählten Telefonnummern für dieses Meeting zu sehen
https://meet.jit.si/static/dialInInfo.html?room=roomname Klicken Sie auf den folgenden Link, um dem Meeting beizutreten:
https://meet.neanderfunk.de/roomname
```

List of currently embedded links of the text:
1. https://meet.jit.si/roomname
2. https://meet.jit.si/static/dialInInfo.html?room=roomname
3. https://meet.neanderfunk.de/roomname

The last one is already correct, but especially the first one is crucial for the UX, but the second dial in link has to be corrected too.

---

## Expected Behavior
Links should have the correct hostname of the configured server URL

```
Klicken Sie auf den folgenden Link, um dem Meeting beizutreten:
https://meet.neanderfunk.de/roomname Wollen Sie sich nur auf Ihrem Telefon einwählen?

Klicken Sie auf diesen Link, um die eingewählten Telefonnummern für dieses Meeting zu sehen
https://meet.neanderfunk.de/static/dialInInfo.html?room=roomname Klicken Sie auf den folgenden Link, um dem Meeting beizutreten:
https://meet.neanderfunk.de/roomname
```

List of links that should be embedded in the text, e.g. if the Server URL is cobfigured to "meet.neanderfunk.de":
1. https://meet.neanderfunk.de/roomname
2. https://meet.neanderfunk.de/static/dialInInfo.html?room=roomname
3. https://meet.neanderfunk.de/roomname

Additionally there is one curly bracket "}" to much in the german text. Don't know why and if this is the case in other languages too, but this is not so important for the UX of inexperienced users.

Note: I'm not related with neanderfunk. I just had a call today and picked a random server to reduce the load of the main servers.

---

## Possible Solution
Use a variable representing the currently configured server URL, like already done in the last link of the invite text.

Doesn't sound to complicated to fix. IMO this is technically not a blocker but regading all the new inexperienced people trying Jitsi on different Servers these days, I consider this high priority regarding the User Experience.

---

## Steps to reproduce
---

1. Open Jitsi Meet App
2. Open Settings ("Einstellungen" in german)
3. Change "Server URL" to something different e.g. "meet.neanderfunk.de"
4. Go Back and open a new conference room e.g. "roomname"
5. Enter the room and click "Invite others"
6. Wait for the sharing dialog to open and choose a app e.g. Signal, Conversations, E-Mail etc.
7. Choose the Contact and there you can see the text mentioned above in ticket.

# Environment details
---

Jitsi Meet Version 20.1.0 on a Wileyfox Swift mobile phone with Android 7.1.2

Don't remember if I installed via F-Droid or Play Store but I think it was F-Droid.

**Comment by saghul** [comment]
Thanks for the report!

**Comment by streetcoder-pm** [comment]
NP. Ok, I'm not an App developer but I tried to look around a bit.

What I've found are the following two code locations, but it doesn't look as simple as I thought. There are already variables used.

List of vars I found:
1. link https://meet.jit.si/roomname seems to be share.mainText {{roomUrl}}
2. link https://meet.jit.si/static/dialInInfo.html?room=roomname seems to be share.dialInfoText {{dialInfoPageUrl}}
3. link https://meet.neanderfunk.de/roomname seems to be the same one as the first link share.mainText {{roomUrl}}
4. and the unexpected curly bracket "}" may be related to share.dialInfoText {{{defaultDialInNumber}}}

I wonder why the {{roomUrl}} doesn't get replaced in the first link. Maybe that is already set during initialization and the later one replaced during runtime, but I'm just guessing. Don't know how the code works.

Translations are in this file but the share.mainText seems to be used used twice in another function:
file https://github.com/jitsi/jitsi-meet/blob/e0afd8fadb706efb21f2f47b6044ab6d5a00312c/lang/main-de.json line 558 ff
```
    "share": {
        "dialInfoText": "\n\n=====\n\nWollen Sie sich nur auf Ihrem Telefon einwählen?\n\n{{{defaultDialInNumber}}}Klicken Sie auf diesen Link, um die eingewählten Telefonnummern für dieses Meeting zu sehen\n{{dialInfoPageUrl}}",
        "mainText": "Klicken Sie auf den folgenden Link, um dem Meeting beizutreten:\n{{roomUrl}}"
    },
```

Here is some function which might be related, but I don't understand how it works:
file https://github.com/jitsi/jitsi-meet/blob/ed5351d25023cc5cf95ef6a42254a0ff0266b0da/react/features/invite/functions.js 
line 411 ff
`    let infoText = i18next.t('share.mainText', { roomUrl });`

line 473 ff
```
infoText += i18next.t('share.dialInfoText', {
                    defaultDialInNumber,
                    dialInfoPageUrl });
```

May be the strings are concatenated and replaced differently during runtime. But by just doing a short code review I don't understand how share.mainText is may be used twice or how and when {{roomUrl}} is replaced differently.

Hmm... maybe my info helps somebody to trace this down, my knowledge ends here ...

**Comment by monofox** [comment]
Tested several cases with the android app Jitsi-Meet 20.1.0 build 201000 given on F-Droid in different languages and different (non)custom servers. Also given meet.neanderfunk.de as server url in android Jitsi-Meet App results in correct data.
Tested also selecting another room which do not fit to the server url in settings anymore (e.g. configured meet.neanderfunk.de but have one from another server in history) results in correct link.
So not able to reproduce. 

@streetcoder-pm : does it still happen after restart of application after changing server url?

**Comment by streetcoder-pm** [comment]
Hi @monofox ,
thanks for testing this out. I tested it again and it didn't happen at all after restarting the app. I did some additional tests with just changing the Server URL and not restarting and now it only happened one time out of six. 

I'm not sure if this case was related to the way my phone is handling the clipboard with previous sharing attempts. May be just my phone or chat app hangs sometimes, it's a rather old one.

I propose to close the ticket until somebody else experiences this behavior too. After the code review and retesting I don't think that the issue is caused by the jitsi app itself.

Thanks for all the effort you put in and sorry for confusion.
